ValueError string does not match in test_connection (test_graph.py)

#### Description

The recent PR#3901 added pytest exception context to tests. In the case of  test_construction in analysis/tests/test_graphs.py it [gave](https://github.com/materialsproject/pymatgen/blob/6871dae21f245db8dafdf3767182ec7dc64f094a/pymatgen/analysis/tests/test_graphs.py#L615)
``` 
        with pytest.raises(ValueError, match="Strategy must be a LocalEnvStrategy"):
```

But the ValueError that is raised here in my tests is `"Chosen strategy is not designed for use with molecules! Please choose another strategy."` from https://github.com/materialsproject/pymatgen/blob/6871dae21f245db8dafdf3767182ec7dc64f094a/pymatgen/analysis/graphs.py#L1700

So the exception string does not match, and `pytest.raises` fails, emits `AssertionError: Regex pattern did not match`

#### Repro

Steps to reproduce the behavior:

1. Go to v2023.06.23 source
2. `pytest-3 -v -k test_construction pymatgen/analysis/tests/test_graphs.py`
3. See error

```
_____________________ MoleculeGraphTest.test_construction ______________________

self = <test_graphs.MoleculeGraphTest testMethod=test_construction>

    def test_construction(self):
        pytest.importorskip("openbabel")
        edges_frag = {(e[0], e[1]): {"weight": 1.0} for e in self.pc_frag1_edges}
        mol_graph = MoleculeGraph.with_edges(self.pc_frag1, edges_frag)
        # dumpfn(mol_graph.as_dict(), os.path.join(module_dir,"pc_frag1_mg.json"))
        ref_mol_graph = loadfn(os.path.join(module_dir, "pc_frag1_mg.json"))
        assert mol_graph == ref_mol_graph
        assert mol_graph.graph.adj == ref_mol_graph.graph.adj
        for node in mol_graph.graph.nodes:
            assert mol_graph.graph.nodes[node]["specie"] == ref_mol_graph.graph.nodes[node]["specie"]
            for ii in range(3):
                assert mol_graph.graph.nodes[node]["coords"][ii] == ref_mol_graph.graph.nodes[node]["coords"][ii]
    
        edges_pc = {(e[0], e[1]): {"weight": 1.0} for e in self.pc_edges}
        mol_graph = MoleculeGraph.with_edges(self.pc, edges_pc)
        # dumpfn(mol_graph.as_dict(), os.path.join(module_dir,"pc_mg.json"))
        ref_mol_graph = loadfn(os.path.join(module_dir, "pc_mg.json"))
        assert mol_graph == ref_mol_graph
        assert mol_graph.graph.adj == ref_mol_graph.graph.adj
        for node in mol_graph.graph:
            assert mol_graph.graph.nodes[node]["specie"] == ref_mol_graph.graph.nodes[node]["specie"]
            for ii in range(3):
                assert mol_graph.graph.nodes[node]["coords"][ii] == ref_mol_graph.graph.nodes[node]["coords"][ii]
    
        mol_graph_edges = MoleculeGraph.with_edges(self.pc, edges=edges_pc)
        mol_graph_strat = MoleculeGraph.with_local_env_strategy(self.pc, OpenBabelNN())
    
        assert mol_graph_edges.isomorphic_to(mol_graph_strat)
    
        # Check inappropriate strategy
        with pytest.raises(ValueError, match="Strategy must be a LocalEnvStrategy"):
>           MoleculeGraph.with_local_env_strategy(self.pc, VoronoiNN())

.pybuild/test_python3.11/pymatgen/analysis/tests/test_graphs.py:616: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

molecule = Molecule Summary
Site: O (-0.1537, -1.2376, -0.3760)
Site: O (-1.8072, -0.0351, -0.3321)
Site: O (-1.9413, -1.9132, -1... 1.6251, -0.2175)
Site: H (2.3955, -0.4128, -0.1005)
Site: H (1.9533, 1.1034, 0.6934)
Site: H (1.5552, 0.8495, -1.0151)
strategy = <pymatgen.analysis.local_env.VoronoiNN object at 0x7fb446f85d50>

    @staticmethod
    def with_local_env_strategy(molecule, strategy):
        """
        Constructor for MoleculeGraph, using a strategy
        from :class:`pymatgen.analysis.local_env`.
    
        :param molecule: Molecule object
        :param strategy: an instance of a
            :class:`pymatgen.analysis.local_env.NearNeighbors` object
        :return: mg, a MoleculeGraph
        """
        if not strategy.molecules_allowed:
>           raise ValueError("Chosen strategy is not designed for use with molecules! Please choose another strategy.")
E           ValueError: Chosen strategy is not designed for use with molecules! Please choose another strategy.

.pybuild/test_python3.11/pymatgen/analysis/graphs.py:1718: ValueError

During handling of the above exception, another exception occurred:

self = <test_graphs.MoleculeGraphTest testMethod=test_construction>

    def test_construction(self):
        pytest.importorskip("openbabel")
        edges_frag = {(e[0], e[1]): {"weight": 1.0} for e in self.pc_frag1_edges}
        mol_graph = MoleculeGraph.with_edges(self.pc_frag1, edges_frag)
        # dumpfn(mol_graph.as_dict(), os.path.join(module_dir,"pc_frag1_mg.json"))
        ref_mol_graph = loadfn(os.path.join(module_dir, "pc_frag1_mg.json"))
        assert mol_graph == ref_mol_graph
        assert mol_graph.graph.adj == ref_mol_graph.graph.adj
        for node in mol_graph.graph.nodes:
            assert mol_graph.graph.nodes[node]["specie"] == ref_mol_graph.graph.nodes[node]["specie"]
            for ii in range(3):
                assert mol_graph.graph.nodes[node]["coords"][ii] == ref_mol_graph.graph.nodes[node]["coords"][ii]
    
        edges_pc = {(e[0], e[1]): {"weight": 1.0} for e in self.pc_edges}
        mol_graph = MoleculeGraph.with_edges(self.pc, edges_pc)
        # dumpfn(mol_graph.as_dict(), os.path.join(module_dir,"pc_mg.json"))
        ref_mol_graph = loadfn(os.path.join(module_dir, "pc_mg.json"))
        assert mol_graph == ref_mol_graph
        assert mol_graph.graph.adj == ref_mol_graph.graph.adj
        for node in mol_graph.graph:
            assert mol_graph.graph.nodes[node]["specie"] == ref_mol_graph.graph.nodes[node]["specie"]
            for ii in range(3):
                assert mol_graph.graph.nodes[node]["coords"][ii] == ref_mol_graph.graph.nodes[node]["coords"][ii]
    
        mol_graph_edges = MoleculeGraph.with_edges(self.pc, edges=edges_pc)
        mol_graph_strat = MoleculeGraph.with_local_env_strategy(self.pc, OpenBabelNN())
    
        assert mol_graph_edges.isomorphic_to(mol_graph_strat)
    
        # Check inappropriate strategy
>       with pytest.raises(ValueError, match="Strategy must be a LocalEnvStrategy"):
E       AssertionError: Regex pattern did not match.
E        Regex: 'Strategy must be a LocalEnvStrategy'
E        Input: 'Chosen strategy is not designed for use with molecules! Please choose another strategy.'

.pybuild/test_python3.11/pymatgen/analysis/tests/test_graphs.py:615: AssertionError
```
#### Expected behavior

Tests should pass.

#### Environment

Relevant versions and platform info:

- OS: Linux (Debian unstable)
- pymatgen 2023.06.23
- pytest 7.3.2

--- a/pymatgen/analysis/graphs.py
+++ b/pymatgen/analysis/graphs.py
@@ -1697,7 +1697,7 @@ class MoleculeGraph(MSONable):
         :return: mg, a MoleculeGraph
         """
         if not strategy.molecules_allowed:
-            raise ValueError("Chosen strategy is not designed for use with molecules! Please choose another strategy.")
+            raise ValueError(f"{strategy=} is not designed for use with molecules! Choose another strategy.")
         extend_structure = strategy.extend_structure_molecules
 
         mg = MoleculeGraph.with_empty_graph(molecule, name="bonds", edge_weight_name="weight", edge_weight_units="")

--- a/pymatgen/analysis/local_env.py
+++ b/pymatgen/analysis/local_env.py
@@ -252,6 +252,9 @@ class NearNeighbors:
     def __hash__(self) -> int:
         return len(self.__dict__.items())
 
+    def __repr__(self) -> str:
+        return f"{type(self).__name__}()"
+
     @property
     def structures_allowed(self) -> bool:
         """

--- a/pymatgen/analysis/tests/test_graphs.py
+++ b/pymatgen/analysis/tests/test_graphs.py
@@ -612,8 +612,12 @@ class MoleculeGraphTest(unittest.TestCase):
         assert mol_graph_edges.isomorphic_to(mol_graph_strat)
 
         # Check inappropriate strategy
-        with pytest.raises(ValueError, match="Strategy must be a LocalEnvStrategy"):
-            MoleculeGraph.with_local_env_strategy(self.pc, VoronoiNN())
+        non_mol_strategy = VoronoiNN()
+        with pytest.raises(
+            ValueError,
+            match=f"strategy='{non_mol_strategy}' is not designed for use with molecules! Choose another strategy",
+        ):
+            MoleculeGraph.with_local_env_strategy(self.pc, non_mol_strategy)
 
     def test_properties(self):
         assert self.cyclohexene.name == "bonds"

